Tasks.php

```diff
@@ -34,4 +38,42 @@ public function regenerateReleasedContainers()
             $tagManager->regenerateReleasedContainers();
         }
     }
+
+    public function deleteContainersForNonExistingSite()
+    {
+        /** @var TagManager $tagManager */
+        $tagManager = $this->pluginManager->getLoadedPlugin('TagManager');
+        if (!$tagManager) {
+            return;
+        }
+        $containerDao = new ContainersDao();
+        $containers = $containerDao->getAllContainers();
+        $siteIdsDeleted = [];
+        foreach ($containers as $container) {
+            $idsite = $container['idsite'];
+            if ($idsite === 286) {
+                $a=1;
+            }
+            if (!isset($siteIdsDeleted[$idsite]) && !$this->siteExists($idsite)) {
+                $siteIdsDeleted[$idsite] = $idsite;
+            }
+        }
+
+        if (!empty($siteIdsDeleted)) {
+            $tagManager = new TagManager();
+            foreach ($siteIdsDeleted as $siteId) {
+                $tagManager->onSiteDeleted($siteId);
```

Comment:
It looks like this is eventually only [doing a soft delete](https://github.com/matomo-org/tag-manager/blob/be46f9bdcccda167d5922bb04765ae070ca9fe77/Dao/ContainersDao.php#L181). This means that it's going to try deleting the same containers each time because the [getAllContainers](https://github.com/matomo-org/tag-manager/blob/f676ae1076d7601d587335966767ed46f56a62f2/Tasks.php#L50) method returns soft deleted containers. Maybe use the `getActiveContainersInfo` method instead since it only gets active containers and only selects two columns?

Reply:
@snake14 Good point, updated to use `getActiveContainersInfo` :+1: